### DISCUSSION TITLE
Question: What is the custom native code that prevents from running on Expo Go?

### DISCUSSION THREAD

**Discussion by tsuz** [rnmapbox/maps#1671] [Q&A]
**Why**

I am trying to install this library so that it's compatible with Expo Go (to support [`expo build`][6]) but in the [installation method][1], it says 

> This package cannot be used in the "Expo Go" app because it requires custom native code.`

Can someone point to which custom native code is required? I thought it was only `pre_install/post_install` ([doc][2], [file][5]) but this doesn't seem to be required [according to Mapbox docs][3]. I believe I'm missing something else here.

FWIW, I've taken a look at a related PR [here][4] but no luck.

Thanks,

[1]: https://github.com/react-native-mapbox-gl/maps/blob/master/plugin/install.md#expo-installation
[2]: https://github.com/react-native-mapbox-gl/maps/blob/master/ios/install.md#react-native--0600
[3]: https://docs.mapbox.com/ios/maps/guides/install/#add-the-dependency
[4]: https://github.com/react-native-mapbox-gl/maps/pull/1388
[5]: https://github.com/EvanBacon/maps/blob/22b4333f906433b5c8d7819bb65b6b944e585f2b/plugin/src/withMapbox.ts
[6]: https://docs.expo.dev/workflow/customizing/#releasing-apps-with-custom-native-code-to

**Comment by BelyaninD** [comment]
You need to build your dev client using EAS Build.

https://docs.expo.dev/development/getting-started/

**Reply by tsuz** [reply]
@BelyaninD 

Thanks for the answer. 
My question was probably unclear - to rephrase it, "why do we need to build my dev client for Mapbox but not for Google Maps?"

**Reply by BelyaninD** [reply]
I think because Expo Go contains all Expo packages. Among them there is react-native-maps, which implements Google Maps using google services for android and iMap(?) on ios.

**Comment by wodin** [comment]
@tsuz, an Expo custom dev client is basically a custom version of Expo Go with your dependencies installed.
https://docs.expo.dev/development/introduction/

As @BelyaninD says, Expo Go includes all the native code needed by the Expo SDK. If you want to use any other dependencies that need native code that is not included in the Expo SDK, then you'll need to build a dev client and use that instead of Expo Go.

`rnmapbox/maps` includes native code that is not included in the Expo SDK, so if you want to use it, you'll need to build a dev client.

> Can someone point to which custom native code is required?

It's all the code under [`android`](https://github.com/rnmapbox/maps/tree/main/android) and [`ios`](https://github.com/rnmapbox/maps/tree/main/ios)
